libmachine/provision/configure_swarm.go

```diff
@@ -67,60 +31,75 @@ func configureSwarm(p Provisioner, swarmOptions swarm.Options, authOptions auth.
 	parts := strings.Split(u.Host, ":")
 	port := parts[1]
 
+	dockerPort := "2376"
 	dockerDir := p.GetDockerOptionsDir()
+	dockerHost := fmt.Sprintf("tcp://%s:%s", ip, dockerPort)
+	dockerClient := DockerClient{dockerHost, authOptions}
+	advertiseInfo := fmt.Sprintf("%s:%s", ip, dockerPort)
 
-	swarmCmdContext := SwarmCommandContext{
-		ContainerName: "",
-		Env:           swarmOptions.Env,
-		DockerDir:     dockerDir,
-		DockerPort:    2376,
-		IP:            ip,
-		Port:          port,
-		AuthOptions:   authOptions,
-		SwarmOptions:  swarmOptions,
-		SwarmImage:    swarmOptions.Image,
-	}
+	if swarmOptions.Master {
 
-	// First things first, get the swarm image.
-	if _, err := p.SSHCommand(fmt.Sprintf("sudo docker pull %s", swarmOptions.Image)); err != nil {
-		return err
-	}
+		cmd := fmt.Sprintf("manage --tlsverify --tlscacert=%s --tlscert=%s --tlskey=%s -H %s --strategy %s --advertise %s",
+			authOptions.CaCertRemotePath,
+			authOptions.ServerCertRemotePath,
+			authOptions.ServerKeyRemotePath,
+			swarmOptions.Host,
+			swarmOptions.Strategy,
+			advertiseInfo)
+
+		cmdMaster := strings.Fields(cmd)
+		for _, option := range swarmOptions.ArbitraryFlags {
+			cmdMaster = append(cmdMaster, "--"+option)
+		}
 
-	swarmMasterCmdTemplate := `sudo docker run -d \
---restart=always \
---net=host \
-{{range .Env}} -e {{.}}{{end}} \
---name swarm-agent-master \
--p {{.Port}}:{{.Port}} \
--v {{.DockerDir}}:{{.DockerDir}} \
-{{.SwarmImage}} \
-manage \
---tlsverify \
---tlscacert={{.AuthOptions.CaCertRemotePath}} \
---tlscert={{.AuthOptions.ServerCertRemotePath}} \
---tlskey={{.AuthOptions.ServerKeyRemotePath}} \
--H {{.SwarmOptions.Host}} \
---strategy {{.SwarmOptions.Strategy}} {{range .SwarmOptions.ArbitraryFlags}} --{{.}}{{end}} {{.SwarmOptions.Discovery}}
-`
-
-	swarmWorkerCmdTemplate := `sudo docker run -d \
---restart=always \
---net=host \
-{{range .Env}} -e {{.}}{{end}} \
---name swarm-agent \
-{{.SwarmImage}} \
-join --advertise {{.IP}}:{{.DockerPort}} {{.SwarmOptions.Discovery}}
-`
+		//Discovery must be at end of command
+		cmdMaster = append(cmdMaster, swarmOptions.Discovery)
 
-	if swarmOptions.Master {
-		log.Debug("Launching swarm master")
-		if err := runSwarmCommandFromTemplate(p, swarmMasterCmdTemplate, swarmCmdContext); err != nil {
+		hostBind := fmt.Sprintf("%s:%s", dockerDir, dockerDir)
+		masterHostConfig := dockerclient.HostConfig{
+			RestartPolicy: dockerclient.RestartPolicy{"Always", 0},
```

Comment:
`go vet` is complaining that you did not say which struct literal field `"Always"` and `0` are meant to belong to here.